tests/test_stdiomgr_base.py

```diff
@@ -46,21 +46,54 @@ def test_capture_stdout(convert_newlines):
 
 
 def test_capture_stderr(convert_newlines, skip_warnings):
```

Comment:
Probably best to augment the name of this test, to indicate it's testing `catch_warnings`? `test_capture_stderr_catchwarnings`, say.

Reply:
Then i need to update the other tests which reuse this method.  Happy to do it. Just saying.

Reply:
Hrmph, good point. Don't rename.

Basic question, though: since this test gets reused so much, does it make more sense to have all of the reuse involve this catch_warnings machinery? Or would it be better to have all the reuse be over, e.g., the simpler testing logic of [`test_capture_stderr_warn`](https://github.com/bskinn/stdio-mgr/pull/52/files/f30f5970ce5deba5dd0651b069c2eff4f929459a#diff-39cba99b32d30ce6f89f7f0709031ff0R83)?

I could see it going either way.

Reply:
IMO it doesnt add any benefits - catch_warnings is a specific usage, and complicated enough stdlib function that is worth a dedicated test, but after the `.warn` weaves through the warnings module, it is eventually just doing `sys.stderr.write` same as `print`.  It is the weaving through warnings module that is worth testing; not its eventual interaction with `sys.stderr`.